deps/rabbit/src/rabbit_quorum_queue.erl

```diff
@@ -315,29 +315,27 @@ declare_queue_error(Error, Queue, Leader, ActingUser) ->
 ra_machine(Q) ->
     {module, rabbit_fifo, ra_machine_config(Q)}.
 
-ra_machine_config(Q) when ?is_amqqueue(Q) ->
+gather_policy_config(Q, ShouldLog) ->
     QName = amqqueue:get_name(Q),
-    {Name, _} = amqqueue:get_pid(Q),
     %% take the minimum value of the policy and the queue arg if present
     MaxLength = args_policy_lookup(<<"max-length">>, fun min/2, Q),
     OverflowBin = args_policy_lookup(<<"overflow">>, fun policy_has_precedence/2, Q),
-    Overflow = overflow(OverflowBin, drop_head, QName),
+    Overflow = overflow(OverflowBin, drop_head, QName, ShouldLog),
     MaxBytes = args_policy_lookup(<<"max-length-bytes">>, fun min/2, Q),
     DeliveryLimit = case args_policy_lookup(<<"delivery-limit">>,
                                             fun resolve_delivery_limit/2, Q) of
                         undefined ->
-                            rabbit_log:info("~ts: delivery_limit not set, defaulting to ~b",
-                                             [rabbit_misc:rs(QName), ?DEFAULT_DELIVERY_LIMIT]),
+                            maybe_log(ShouldLog, info,
+                                      "~ts: delivery_limit not set, defaulting to ~b",
```

Comment:
dear Core Team, should this message be logged unconditionally as well? This is not a misconfiguration, if a user is happy with the default value and does not set an explicit delivery-limit, this will be logged all the time for all the quorum queues.

Reply:
It's there to be clear that there is a default set. We can perhaps lower it to debug in 4.1 or remove it completely but I think making users aware of this potentially breaking change doesn't harm.

Reply:
We can move this to a function that is not called periodically and remove it from this function.

Reply:
In fact, we don't necessarily have to do it in `rabbit_quorum_queue` if there's a more suitable alternative where the delivery limit is known.